examples/test_util/EclIO.cpp

```diff
@@ -0,0 +1,695 @@
+ 
+        if (dhead!=dtail){ 
+            std::string message="Error reading binary doub data, tail not matching header.";
+            throw std::runtime_error(message);
+        }
+    } 
+   
+    return arr;
+}
+
+std::vector<bool> EclIO::readBinaryLogiArray(std::fstream &fileH,const int size){
+
+    std::vector<bool> arr;
+    bool value;
+    int num,dhead,dtail,rest;
+    unsigned int intVal;
+
+    if (!fileH.is_open()){
+        std::string message="fstream fileH not open for reading";
+        throw std::runtime_error(message);
+    }
+   
+    rest=size;
+   
+    while (rest > 0) {   
+ 
+        fileH.read((char *)&dhead, sizeof(dhead));
+        dhead=flipEndian<int>(dhead);
+   
+        num=dhead/EIOD::sizeOfLogi;
+
+        if ((num > EIOD::MaxNumLogi) || (num < 0)){
+            std::string message="Error reading binary logi data, inconsistent header data or incorrect number of elements";
+	    throw std::runtime_error(message);
+        }
+
+        for (int i=0;i<num;i++){
+
+            fileH.read((char *)&intVal, EIOD::sizeOfLogi);
+           
+            if (intVal==0xffffffff){
+                value=true;
+            } else if (intVal==0x00000000){
+                value=false;
+            } else {
+                std::string message="Error reading log value from element " + std::to_string(i);
+                throw std::runtime_error(message);
+            }
+           
+            arr.push_back(value);
+        }
+   
+        rest=rest-num;
+
+        if (((num < EIOD::MaxNumLogi) && (rest!=0)) || ((num==EIOD::MaxNumLogi) && (rest < 0))) {
+            std::string message="Error reading binary logi data, incorrect number of elements";
+            throw std::runtime_error(message);
+        }
+       
+        fileH.read((char *)&dtail, sizeof(dtail));
+        dtail=flipEndian<int>(dtail);
+ 
+        if (dhead!=dtail){ 
+            std::string message="Error reading binary logi data, tail not matching header.";
+            throw std::runtime_error(message);
+        }
+    } 
+   
+    return arr;
+}
+
+std::vector<std::string> EclIO::readBinaryCharArray(std::fstream &fileH,const int size){
+
+    std::string str8(8,' ');
+    std::vector<std::string> arr;
+   
+    int num,dhead,dtail,rest;
+   
+    if (!fileH.is_open()){
+        std::string message="fstream fileH not open for reading";
+        throw std::runtime_error(message);
+    }
+
+    rest=size;
+   
+    while (rest > 0) {   
+ 
+        fileH.read((char *)&dhead, sizeof(dhead));
+        dhead=flipEndian<int>(dhead);
+        
+        num=dhead/EIOD::sizeOfChar;
+
+        if ((num > EIOD::MaxNumChar) || (num < 0)){
+            std::string message="Error reading binary char data, inconsistent header data or incorrect number of elements";
+	        throw std::runtime_error(message);
+        }
+
+        for (int i=0;i<num;i++){
+            fileH.read(&str8[0], EIOD::sizeOfChar);
+            arr.push_back(str8);
```

Comment:
This will internalize the strings *with* trailing spaces up to length 8; these trailing spaces should be removed before storing the strings in the `std::string`.

Reply:
I'm removing trailing spaces for array name in the constructor for EclFile before these are stored in the data structure of EclFile. This will make it easier when looking up vectors. Example getVect("PERMX").

When it comes to arrays of strings (actual char data). Why would you argue that these should be stored with out trailing spaces ?

Reply:
fixed, trailing spaces are removed